Refactor Modals

colbymorrison opened the pull request:
Closes #300 
Closes #306 

"Componentizes" modals and refactors their CSS

- All modals have a title, so I added it as a prop to `Modal.js`
- `modal.scss` now contains styling for modal titles and PendingOperationButtons, so their margins are consistent across modals. It also now has styling for modal width and padding, so those are also consistent. 
- Creates a new component called `ModalInput` to encapsulate a `label` and `input` in a Modal, such as this:
![modal_input](https://user-images.githubusercontent.com/20446774/88451537-3f96d380-ce0c-11ea-9079-816edbe326ab.png)
Its props include: if the input is required, a function to check the input is valid, and a message to display under the text box if the input is invalid. See the comment in `ModalInput.js` for a description of all props. `modal_input.scss` allows us to change the styling for `ModalInput`s in one place. This really cleans up the modal css iles.
- Creates a new component `CancelConfirmButtons` for use in `ChangeRoleModal` and `ChangeStatusModal` (if we implement the updated add user flow in the Figma, this can be used there too)

colbymorrison commented:
Updated:

1) Added a div as you suggested to not push elements down on error text. Example below for Forgot Password and Add Member modals:
![Screen Capture_Navigator_20200727213734](https://user-images.githubusercontent.com/20446774/88620144-21221980-d052-11ea-9bdb-57d791d92221.gif)

2) Reset image modal:
![reset](https://user-images.githubusercontent.com/20446774/88620087-f6d05c00-d051-11ea-8230-5cff7bb114fb.png)

3) Change password modal:
![reset-pass](https://user-images.githubusercontent.com/20446774/88620099-02bc1e00-d052-11ea-9cc2-1537e99f897f.png)

jcolla-holla commented:
Nice.  I see a few more things that need addressing:

- [ ] The "Uploading..." text and loading graphic is not centered on uploading a picture.  And the original modal header remains.  I think it's ok if the original header remains but it may be better to remove it in favor of only "Uploading...".

<img width="594" alt="Screen Shot 2020-07-27 at 10 26 01 PM" src="https://user-images.githubusercontent.com/16062590/88622789-8b968200-d071-11ea-8359-2650e5334822.png">

- [ ] The `font-weight` of modal buttons (at least a few I looked at) is `bold` but in other cases buttons have a value of `600`.  `bold` is a value of `700` so there is a slight inconsistency between modals and other buttons in the app.  Let's have all buttons have a `font-weight` of `600` (I believe this is in line w Figma)

colbymorrison commented:
Updated "Uploading" Screen:
![uploading](https://user-images.githubusercontent.com/20446774/88739742-20d85b80-d0f0-11ea-8ff1-990febb1da58.png)

Changed default font weight for buttons to 600 in `modal.scss`

colbymorrison commented:
Updated the button style on that Modal to `save-button`, the same as the other modals
![pass](https://user-images.githubusercontent.com/20446774/88757490-0e274c00-d11b-11ea-9693-586c66fa3400.png)

ibeckermayer commented on frontend/client/src/components/Modal.js:
```diff
@@ -19,6 +24,7 @@ const Modal = (props) => {
           </span>
         )}
+        <h3>{props.title}</h3>
         {props.children}
```
I think we can remove this, doesn't seem to be used anywhere else

ibeckermayer replied:
I was wrong, it turns out `props.children` is a special value, see [here](https://reactgo.com/react-props-children/)

ibeckermayer commented on frontend/client/src/components/ModalInput.js:
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+
+/**
```
Good documentation

jcolla-holla commented:
> Great job, ready to merge assuming all of @jcolla-holla's critiques are accounted for. Just requested to remove an unused line (see my inline comments)

Yup, all addressed.  Looks like the final touches are what your comments point out.
